### DISCUSSION TITLE
Fire an event on execuition of db->dropAllTables();

### DISCUSSION THREAD

**Discussion by realtebo** [laravel/framework#42858] [Ideas]
*Whene developing*, we need to intercept when db is refreshed / wiped.

I ask you kindly to add firing of a 'db_wipe' event or similar.

So we can intercept and,. for example, be sure to dump some not rebuildable data from db to file.

**Comment by henzeb** [comment]
Your approach is wrong. 
1. You need to drop the privileges so that the user used for connecting your laravel app to the database has no right to drop anything.
2. use a second user for migrations which is able to drop a table (preferably in a CI like setting)
3. you should have regular backups
4. you should use binlogs that can be executed starting from at least the last full backup until the database crash/wipe.

**Comment by realtebo** [comment]
Ok, I understand your point of view. Actually in production, a part from shell user, there is no way to drop all tables.

Error of mine: I forgot to precise that we need this for help in development.

We have some 'fixed' data needed to be reimported, so MUST survive to db:wipe as a file dump.

actually we are using mysql dump to dump files, then wiping or doing a migrate:fresh, then reimporting using mysql console command

We would like to be able to intercept wipe request to act FROM CODE to force a dump of some tables. To understanda the point, these tables are html content of email templates and some fixed pages. So we must have always latest version in a file (and we will commit this until first release)

We cannot use a seed to import that data from a file because the developers are actively changing the data itself and so we need to help developers be sure to not loose last versions of data they edited / received / modified

**Reply by henzeb** [reply]
I have dealt with a similar situation before. templates of pages and e-mails were done by frontend-developers, we just used a skeleton template or a draft given by them to us to see if all the data comes through.

But if you must, you could replace the commands with your own version of it: 

````php
 $this->app->extend(Illuminate\Database\Console\WipeCommand::class, function(){
    //...
});
````
